mod_form.php

```diff
@@ -121,10 +121,14 @@ public function definition() {
 
             foreach ($mycourses as $mycourse) {
                 $courselabel = $catlist[$mycourse->category] . ' / ' . $mycourse->fullname.' ('.$mycourse->shortname.')';
-                $options[$mycourse->id] = $courselabel;
                 if (empty($mycourse->visible)) {
-                    $hiddenlabel = ' '.get_string('hiddencourse', 'subcourse');
-                    $options[$mycourse->id] .= $hiddenlabel;
+                    if($config->displayhiddencourses || $mycourse->id == $currentrefcourseid){
+                        $hiddenlabel = ' '.get_string('hiddencourse', 'subcourse');
+                        $options[$mycourse->id] =  $courselabel.$hiddenlabel;
```

Comment:
this line will trigger codiing guideline failures - the gitlab pipelines might show them, but you might want to investigate installing moodle's codechecker tools locally or integrating them with your IDE as well - do a search on your IDE name (vscode etc) and moodle coding guidelines and you should find how to set it up